docs/GLOSSARY.md

```diff
@@ -0,0 +1,23 @@
+# Glossary
+Snap is simple in scope and it becomes more simple when you know the terminology we use throughout the project. Here they are: 
+
+* **Config: Global Config** - Values loaded at runtime of the daemon ([reference](https://github.com/intelsdi-x/snap/blob/master/docs/SNAPD_CONFIGURATION.md))
+* **Config: Global Options** - Values passed as command-line parameters or environmental variables ([reference](https://github.com/intelsdi-x/snap/blob/f80bf52b8ae93d3b04e3c97761b6fb750174eb50/docs/SNAPCTL.md#global-options))
+* **Config: Metric Config** - key/value pairs shared by collector namespace in a Task Manifest ([example](https://github.com/intelsdi-x/snap-plugin-collector-meminfo/blob/master/examples/tasks/task-mem.json#L15))
+* **Config: Plugin Config** - key/value pairs stored in the `config` block within a Task Manifest ([example](https://github.com/intelsdi-x/snap-plugin-collector-meminfo/blob/master/examples/tasks/task-mem.json#L24))
+* **Metric Catalog** - List of all available metrics exposed by a running instance of the Snap daemon
+* **Plugin** - A [binary][binary] that is compatible with Snap by following the [Plugin Lifecycle](PLUGIN_LIFECYCLE.md)
+* **Plugin Type: Collector** - Uses sensors to gather data and present as a dynamically-generated namespaced catalog
+* **Plugin Type: Processor** - Extends or filters collected metrics
+* **Plugin Type: Publisher** - Persists metrics into a target endpoint
+* **Snap** -  The project name, focused on the Snap daemon and the plugins that power its collection, processing and publishing of telemetry
+* **Snap Telemetry** - The full name of the Snap project, used mostly for easy searching (like snap-telemetry.io) or hashtag (#SnapTelemetry)
+* **`snapctl`** - The command-line interface (CLI) for Snap, released as a [binary][binary]
```

Comment:
I want the name change to happen in the same PR and not mix it up in this one. Anyone changing binary names will change this file via regex.